apis/k8ssandra/v1alpha1/constants.go

```diff
@@ -9,6 +9,25 @@ const (
 	// regardless of whether the replication of the system keyspaces changes.
 	SystemReplicationAnnotation = "k8ssandra.io/system-replication"
```

Comment:
Would it provide a bit more clarity to the code if we renamed this to `InitialSystemReplicationAnnotation`? Indeed afaict this is only used inside `checkSystemReplication` for a newly-created cluster. It doesn't play any role when a new DC is added, correct?

Reply:
The name change would indeed provide more clarity. I'll do that. And yes you are correct that it plays no role when adding a DC to an existing cluster.

For what it's worth I am not at all happy about how we manage replication for system keyspaces, and I want to change it post-1.0. I'll try to summarize here briefly. 

We pass JVM system properties in an env var which is configured in the PodTemplateSpec. The management-api looks for those system properties. They are only applied once. Knowing this, I introduced a number of subtle, non-trivial changes to make sure those system properties don't change. Why? Because changing them changes the PodTemplateSpec which triggers a StatefulSet update, i.e., rolling restart. I want to manage replication for system keyspaces like we do for other ones. In order to do that though, we need to add support for running repairs, namely on `system_auth`. Since that has a small about of data, the management-api equivalent of `nodetool repair -full` should be fine.

Reply:
> Because changing them changes the PodTemplateSpec which triggers a StatefulSet update, i.e., rolling restart.

Ah that bit explains a lot of the motivation here, thanks.